samples/SentimentAnalyzerCustomSkill/cs/Skill/FaceSentimentAnalyzer/FaceSentimentAnalyzerBinding.cs

```diff
@@ -68,68 +68,56 @@ public bool IsFaceFound
-                }
-                else
-                {
-                    return false;
-                }
+                ISkillFeature feature = m_bindingHelper[FaceSentimentAnalyzerConst.SKILL_OUTPUTNAME_FACEBOUNDINGBOXES];
+                var faceBoundingBoxes = (feature.FeatureValue as SkillFeatureTensorFloatValue).GetAsVectorView();
+                return faceBoundingBoxes.Count > 0;
             }
         }
 
         /// <summary>
-        /// Returns the sentiment with the highest score
+        /// Returns the sentiments with the highest score for each face detected
         /// </summary>
         /// <returns></returns>
-        public SentimentType PredominantSentiment
+        public IReadOnlyList<SentimentType> PredominantSentiments
         {
             get
             {
-                SentimentType predominantSentiment = SentimentType.neutral;
-                ISkillFeature feature = null;
-                if (m_bindingHelper.TryGetValue(FaceSentimentAnalyzerConst.SKILL_OUTPUTNAME_FACESENTIMENTSCORES, out feature))
+                var faceSentimentScores = (m_bindingHelper[FaceSentimentAnalyzerConst.SKILL_OUTPUTNAME_FACESENTIMENTSSCORES].FeatureValue as SkillFeatureTensorFloatValue).GetAsVectorView();
+                var predominantSentiments = new List<SentimentType>();
+               
+                for (int i = 0; i < faceSentimentScores.Count; i += ((int)SentimentType.contempt + 1))
                 {
-                    var faceSentimentScores = (feature.FeatureValue as SkillFeatureTensorFloatValue).GetAsVectorView();
-
                     float maxScore = float.MinValue;
-                    for (int i = 0; i < faceSentimentScores.Count; i++)
+                    var predominantSentiment = SentimentType.neutral;
+                    for (int j = 0; j < ((int)SentimentType.contempt + 1); j++)
                     {
-                        if (faceSentimentScores[i] > maxScore)
+                        float score = faceSentimentScores[i + j];
+                        if (score > maxScore)
                         {
-                            predominantSentiment = (SentimentType)i;
-                            maxScore = faceSentimentScores[i];
+                            predominantSentiment = (SentimentType)j;
+                            if (score >= 0.5f) // since scores are softmax, there can't be 2 scores above 0.5, early break opportunity
+                            {
+                                break;
+                            }
+                            maxScore = score;
                         }
                     }
+                    predominantSentiments.Add(predominantSentiment);
                 }
-                
-                return predominantSentiment;
+                return predominantSentiments;
             }
         }
 
         /// <summary>
-        /// Returns the face rectangle
+        /// Returns the bounding box around each detected faces
```

Comment:
each detected face

Reply:
There was somewhere else with the same text that should also be changed but having trouble finding it now :(